### PR TITLE
Line caps and joins #291

### PR DESCRIPTION
**Pull request by milg15**

### CODE REVIEW

**Comment by dabbott** [comment]
I pushed an update that gets the canvas rendering working. The main thing was propagating the border options into `getStrokedPath`, rather than setting them on the Paint. We actually convert every stroke into a new path and then "fill" it, since that lets us do things like gradient strokes.

File: packages/app/src/containers/BorderInspector.tsx [hunk 1]
```diff
@@ -28,12 +42,81 @@ export default memo(function BorderInspector() {
...
+    [dispatch],
+  );
+  const onChangeLineJoinStyle = useCallback(
+    (value: string) => {
+      dispatch('setBorderLineJoin', parseInt(value));
+    },
+    [dispatch],
+  );
+
+  const BorderInspectorExpanded = useMemo(() => {
```
Comment (dabbott) [inline packages/app/src/containers/BorderInspector.tsx:72]: Lowercase `borderInspectorExpanded` cause it's not a component

File: packages/app/src/components/inspector/ArrayController.tsx [hunk 2]
```diff
@@ -30,10 +30,13 @@ interface ArrayControllerProps<Item> {
   title: ReactNode;
   sortable?: boolean;
   reversed?: boolean;
+  expanded?: boolean;
+  expandedContent?: ReactNode;
```
Comment (dabbott) [inline packages/app/src/components/inspector/ArrayController.tsx:34]: For consistency with `renderItem` let's make this a "render prop" . I.e. `renderExpandedContent?: () => ReactNode`.

This also has the small perf benefit of not needing to run unless it's actually expanded and visible.

**Review by dabbott — CHANGES_REQUESTED** [review]
Looks great! A commented on a couple small things. 

FYI I also remembered I removed the `borderOptions.isEnabled` check in my previous change. I don't think there's actually a way to disable border options in Sketch, and I don't think we'll have one either

**Comment by milg15** [comment]
Hopefully, this makes sense

**Comment by dabbott** [comment]
We were still using `expandedContent` on the CheckboxArrayController, so I swapped it for `renderExpandedContent`

**Review by dabbott — APPROVED** [review]

**Pull request merged**